hazelcast/src/main/java/com/hazelcast/cache/impl/nearcache/impl/store/AbstractNearCacheRecordStore.java

```diff
@@ -270,6 +275,10 @@ public V get(K key) {
     public void put(K key, V value) {
         checkAvailable();
 
+        if (!isEvictionEnabled() && maxSizeChecker != null && maxSizeChecker.isReachedToMaxSize()) {
+            return;
```

Comment:
I just saw that I can simplify the check to:
```
        if (!isEvictionEnabled() && evictionChecker.isEvictionRequired()) {
            return;
        }
```
Which will call the same code from within the `MaxSizeEvictionChecker` class. So it's a little less duplication.